web/modules/custom/weather_block/src/Plugin/Block/WeatherBlock.php

```diff
@@ -11,40 +19,144 @@
   id: "weather_block",
   admin_label: new TranslatableMarkup("Weather block"),
 )]
-class WeatherBlock extends BlockBase {
+class WeatherBlock extends BlockBase implements ContainerFactoryPluginInterface {
+  /**
+   * The Config Factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected ConfigFactoryInterface $configFactory;
+  /**
+   * The Logger service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected LoggerChannelFactoryInterface $logger;
+
+  /**
+   * Constructor for WeatherBlock.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory, LoggerChannelFactoryInterface $logger) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $configFactory;
+    $this->logger = $logger;
+  }
+
+  /**
+   * Container for WeatherBlock.
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('logger.factory')
+       );
+  }
 
   /**
    * {@inheritdoc}
    */
   public function build(): array {
-    $api_data = $this->getDataFromAPI();
+    $cities = $this->configuration['cities'];
+    $api_key = $this->configuration['key'];
+    $api_data = $this->getDataFromApi($cities, $api_key);
+
+    if ($api_data === ['no key']) {
+      return [];
+    }
+
+    $firstCity = reset($api_data);
+    $weatherType = $firstCity['weather-type'];
     $formatted_data = $this->formatData($api_data);
 
     return [
       '#theme' => 'weather_block_template',
       '#content' => $formatted_data,
-      '#weather_type' => $api_data,
+      '#weather_type' => $weatherType,
     ];
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $cities = [
+      'Вінниця', 'Дніпро', 'Донецьк', 'Житомир', 'Запоріжжя', 'Івано-Франківськ',
+      'Київ', 'Кропивницький', 'Луганськ', 'Луцьк', 'Львів', 'Миколаїв', 'Одеса',
+      'Полтава', 'Рівне', 'Суми', 'Тернопіль', 'Ужгород', 'Харків', 'Херсон',
+      'Хмельницький', 'Черкаси', 'Чернівці', 'Чернігів', 'Сімферополь',
+    ];
+
+    $form['cities'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Choose a city'),
+      '#options' => array_combine($cities, $cities),
```

Comment:
not like that is very important, but keys should be machine names, so in English without spaces, etc.
That said these strings should also be in English and wrapped with ->t(), and if you want them in Ukrainian, then translate them using Drupal's translation system.

Translating is not that straightforward, so you could leave it like that.

Also, I don't know if API accepts English names, if not then it is fine.